spec/app/github_repo_fetcher_spec.rb

```diff
@@ -18,4 +18,34 @@
       }.to raise_error(StandardError)
     end
   end
+
+  describe "#readme" do
+    it "caches the first response" do
+      repo_name = SecureRandom.uuid
+      api_endpoint = "https://api.github.com/repos/alphagov/#{repo_name}/readme"
+      stubbed_request = stub_request(:get, api_endpoint)
+        .to_return(status: 200, body: '{ "content": {} }', headers: { content_type: "application/json" })
+
+      GitHubRepoFetcher.new.readme(repo_name)
+      GitHubRepoFetcher.new.readme(repo_name)
+      expect(stubbed_request).to have_been_requested.once
+    end
+
+    it "retrieves the README content from the GitHub API response" do
+      repo_name = SecureRandom.uuid
+      response = { "content": "IyB0ZW1wb3JhcnktdGVzdA==\n" }
```

Comment:
Can we explain where this comes from?

Reply:
I've rewritten this to be a bit more self documenting. (But it came from making an API call to https://github.com/ChrisBAshton/temporary-test, which I've since deleted).